app/javascript/crayons/MultiSelectAutocomplete/MultiSelectAutocomplete.jsx

```diff
@@ -64,78 +102,215 @@ export const MultiSelectAutocomplete = ({ labelText, fetchSuggestions }) => {
     editValue,
     activeDescendentIndex,
     ignoreBlur,
+    showMaxSelectionsReached,
   } = state;
 
   const inputRef = useRef(null);
   const inputSizerRef = useRef(null);
   const selectedItemsRef = useRef(null);
+  const popoverRef = useRef(null);
+
+  const allowSelections =
+    !maxSelections || selectedItems.length < maxSelections;
+
+  useEffect(() => {
+    if (defaultValue.length > 0) {
+      dispatch({ type: 'setSelectedItems', payload: defaultValue });
+    }
+  }, [defaultValue]);
 
   const handleInputBlur = () => {
-    // Since the input is sometimes removed and rendered in a new location on blur, it's possible that inputRef.current may be null when we complete this check.
-    const currentValue = inputRef.current ? inputRef.current.value : '';
+    dispatch({ type: 'setShowMaxSelectionsReached', payload: false });
+
+    const {
+      current: { value: currentValue },
+    } = inputRef;
+
     // The input will blur when user selects an option from the dropdown via mouse click. The ignoreBlur boolean lets us know we can ignore this event.
-    if (!ignoreBlur && currentValue !== '') {
-      selectItem({ selectedItem: currentValue, focusInput: false });
+    if (!ignoreBlur && allowSelections && currentValue !== '') {
+      selectByText({ textValue: currentValue, keepSelecting: false });
     } else {
+      exitEditState({ keepSelecting: false });
       dispatch({ type: 'setSuggestions', payload: [] });
+      dispatch({ type: 'setIgnoreBlur', payload: false });
     }
-
-    dispatch({ type: 'setIgnoreBlur', payload: false });
   };
 
   useEffect(() => {
+    // editValue defaults to null when component is first rendered.
+    // This ensures we do not autofocus the input before the user has started interacting with the component.
+    if (editValue === null) return;
+
     const { current: input } = inputRef;
-    if (inputPosition !== null) {
+    if (input && inputPosition !== null) {
+      // Entering 'edit' mode
       resizeInputToContentSize();
 
       input.value = editValue;
       const { length: cursorPosition } = editValue;
+
       input.focus();
       input.setSelectionRange(cursorPosition, cursorPosition);
-    } else {
-      // Remove inline style added to size the input
-      input.style.width = '';
-      input.focus();
+
+      // Trigger the input event to make sure suggestion UI updates correctly
+      const changeEvent = new Event('input');
+      input.dispatchEvent(changeEvent);
     }
   }, [inputPosition, editValue]);
 
+  useEffect(() => {
+    if (activeDescendentIndex !== null) {
+      const activeItem = document.querySelector('[aria-selected="true"]');
```

Comment:
Will there ever only be one element on the entire page with the attribute key/value pair `aria-selected="true"`? If not, use an element of the component instead to apply `querySelector`.

```suggestion
      const activeItem = someRef.current.querySelector('[aria-selected="true"]');
```

Reply:
There _should be_ only one but yeah more robust to query a specific element 👍